src/PolyType/Utilities/AggregatingTypeShapeProvider.cs

```diff
@@ -0,0 +1,53 @@
+﻿using PolyType.Abstractions;
+
+namespace PolyType.Utilities;
+
+/// <summary>
+/// An <see cref="ITypeShapeProvider"/> that returns the first shape found
+/// from iterating over a given list of providers.
+/// </summary>
```

Comment:
Perhaps name-dropping the chain-of-responsibility pattern in a remarks section would help here.